json_serialization_writer.go

```diff
@@ -585,352 +585,112 @@ func (w *JsonSerializationWriter) GetSerializedContent() ([]byte, error) {
-			if ok {
-				err := w.WriteInt64Value(key, pi64v)
-				if err != nil {
-					return err
-				}
-				continue
-			}
-			i64v, ok := value.(int64)
-			if ok {
-				err := w.WriteInt64Value(key, &i64v)
-				if err != nil {
-					return err
-				}
-				continue
-			}
-			pf32v, ok := value.(*float32)
-			if ok {
-				err := w.WriteFloat32Value(key, pf32v)
-				if err != nil {
-					return err
-				}
-				continue
-			}
-			f32v, ok := value.(float32)
-			if ok {
-				err := w.WriteFloat32Value(key, &f32v)
-				if err != nil {
-					return err
-				}
-				continue
-			}
-			pf64v, ok := value.(*float64)
-			if ok {
-				err := w.WriteFloat64Value(key, pf64v)
-				if err != nil {
-					return err
-				}
-				continue
-			}
-			f64v, ok := value.(float64)
-			if ok {
-				err := w.WriteFloat64Value(key, &f64v)
-				if err != nil {
-					return err
-				}
-				continue
-			}
-			puv, ok := value.(*uuid.UUID)
-			if ok {
-				err := w.WriteUUIDValue(key, puv)
-				if err != nil {
-					return err
-				}
-				continue
-			}
-			uv, ok := value.(uuid.UUID)
-			if ok {
-				err := w.WriteUUIDValue(key, &uv)
-				if err != nil {
-					return err
-				}
-				continue
-			}
-			ptv, ok := value.(*time.Time)
-			if ok {
-				err := w.WriteTimeValue(key, ptv)
-				if err != nil {
-					return err
-				}
-				continue
-			}
-			tv, ok := value.(time.Time)
-			if ok {
-				err := w.WriteTimeValue(key, &tv)
-				if err != nil {
-					return err
-				}
-				continue
-			}
-			pdv, ok := value.(*absser.ISODuration)
-			if ok {
-				err := w.WriteISODurationValue(key, pdv)
-				if err != nil {
-					return err
-				}
-				continue
-			}
-			dv, ok := value.(absser.ISODuration)
-			if ok {
-				err := w.WriteISODurationValue(key, &dv)
-				if err != nil {
-					return err
-				}
-				continue
-			}
-			ptov, ok := value.(*absser.TimeOnly)
-			if ok {
-				err := w.WriteTimeOnlyValue(key, ptov)
-				if err != nil {
-					return err
-				}
-				continue
-			}
-			tov, ok := value.(absser.TimeOnly)
-			if ok {
-				err := w.WriteTimeOnlyValue(key, &tov)
-				if err != nil {
-					return err
-				}
-				continue
-			}
-			pdov, ok := value.(*absser.DateOnly)
-			if ok {
-				err := w.WriteDateOnlyValue(key, pdov)
-				if err != nil {
-					return err
-				}
-				continue
-			}
-			dov, ok := value.(absser.DateOnly)
-			if ok {
-				err := w.WriteDateOnlyValue(key, &dov)
-				if err != nil {
-					return err
-				}
-				continue
-			}
-			ba, ok := value.([]byte)
-			if ok {
-				err := w.WriteByteArrayValue(key, ba)
-				if err != nil {
-					return err
-				}
-				continue
+			err := w.writeAdditional(key, value)
+			if err != nil {
+				return err
 			}
-			return fmt.Errorf("unsupported AdditionalData type: %T", value)
 		}
 	}
 	return nil
 }
 
+// writeAdditional Helper function for writing additional data. Feature adds
+// the ability to support certain map types.
+func (w *JsonSerializationWriter) writeAdditional(key string, input any) error {
```

Comment:
now that we're back to a single method, would you mind moving the body back to the public method and removing this private one?

Reply:
Changes committed.